docs/en/02_Developer_Guides/01_Templates/11_Partial_Template_Caching.md

```diff
@@ -0,0 +1,282 @@
+increase redundancy and improve performance.
+[/note]
+
+
+### SSViewer.global_key
+
+Apart from the value provided in `$CacheKey`, every block also uses `SSViewer.global_key` as one of the extra values for generating
+the cache lookup key. It contains regular template syntax.
+
+By default it equals to `'$CurrentReadingMode, $CurrentUser.ID'`.  
+This ensures the current [Versioned](api:SilverStripe\Versioned\Versioned) state and user ID are used.  
+
+As usual, you may override its value via YAML configs. For example:
+
+```yml
+# app/_config/view.yml
+SilverStripe\View\SSViewer:
+  global_key: '$CurrentReadingMode, $CurrentUser.ID, $Locale'
+```
+
+
+### Nested cached blocks
+
+Every nested cache block is processed independently.
+
+Let's consider the following example:
+```ss
+<% cached $PageKey %>
+  <!-- Header -->
+  <% cached $BodyKey %> <!-- Body --> <% end_cached %>
+  <!-- Footer -->
+<% end_cached %>
+```
+
+The template processor will transparently flatten the structure into something similar to the following pseudo-code:
+
+```ss
+<% cached $PageKey %><!-- Header --><% end_block %>
+<% cached $BodyKey %><!-- Body --><% end_cached %>
+<% cached $PageKey %><!-- Footer --><% end_cached %>
+```
+
+[note]
+`$PageKey` is used twice, but evaluated only once per render because of [template object caching](caching/#object-caching).
+[/note]
+
+
+### Uncached
+
+The tag `<% uncached %> ... <% end_uncached %>` disables caching for its content.  
+
+```ss
+<% cached $PageKey %>
+  <!-- Header -->
+  <% uncached %><!-- Body --><% end_uncached %>
+  <!-- Footer -->
+<% end_cached %>
+```
+
+Because of the nested block flattening (see above), it works seamlessly on any level of depth.  
+
+[warning]
+The `uncached` block only works on the lexical level.
+If you have a template that caches content rendering another template with included uncached blocks,
+those will not have any effect on the parent template caching blocks.
+[/warning]
+
+
+### Nesting in FOR and IF blocks
+
+Currently, a cache block cannot be included in `if` and `loop` blocks.  
+The template engine will throw an error letting you know if you've done this.
+
+[note]
+You may often get around this using aggregates or by un-nesting the block.
+
+E.g.
+
+```
+<% cached $LastEdited %>
+  <% loop $Children %>
+      <% cached $LastEdited %>
+          $Name
+      <% end_cached %>
+  <% end_loop %>
+<% end_cached %>
+```
+
+Might be re-written as something like that:
+
+```
+<% cached $LastEdited %>
+    <% cached $AllChildren.max('LastEdited') %>
+        <% loop $Children %>
+            $Name
+        <% end_loop %>
+    <% end_cached %>
+<% end_cached %>
+```
+[/note]
+
+
+### Unless (syntax sugar)
+
+`if` keyword may be swapped with keyword `unless`, which inverts the boolean value evaluation.
+
+[note]
+The two following forms produce the same result
+
+```ss
+<% cached unless $Key %>
+  <!-- "unless $Cond" === "if not $Cond" -->
```

Comment:
This looks like an HTML comment - is this some kind of alternative template syntax?

Reply:
no, this is just an HTML comment :)

Reply:
it's doing more confusing that educating.  I'd just swap out the comment for <!-- Content -->